torch/optim/_device_utils.py

```
@@ -0,0 +1,79 @@
+    supported_devices = ["cuda", "xpu", "hpu"]
+    
+    if device.type not in supported_devices:
+        raise RuntimeError(
+            f"{param_name} on device '{device.type}' does not support capturable "
+            f"optimizer operations. Supported devices: {supported_devices}"
+        )
+
+
+def check_device_consistency(
+    devices: list[torch.device], 
+    operation: str = "optimizer operation"
```

Ruff (0.14.6)

Unused function argument: `operation`

(ARG001)